libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -804,8 +805,12 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
             routeUpdateMutex.withLock {
                 navigationSession.setRoutes(routes)
-                setRoutesToTripSession(routes, legIndex, reason)
-                directionsSession.setRoutes(routes, legIndex, reason)
+                val processedRoutes = setRoutesToTripSession(routes, legIndex, reason)
+                if (processedRoutes.error == null) {
```

Comment:
It seems like this is the right place to distinguish success from error. It looks like the routes should not be set to directionsSession at all or the state becomes inconsistent. I may be wrong tho. Any thoughts?

Reply:
how can the application react on an invalid routes error? 🤔 it seems we need mechanism for that aka `callback`

Reply:
Another option is to invoke `RoutesObserver` with empty list of routes and a new reason called `ROUTES_UPDATE_REASON_IGNORE_INVALID`, for example. Although with callback it's more obvious which routes were ignored.

Reply:
yeah, let's add it

Reply:
Done, could you review please?